Tools/RunTopTagger.h

```diff
@@ -0,0 +1,235 @@
+        
+        // TopTagger
+        //std::cout << "Create TopTagger object" << std::endl;
+        TopTagger tt("TopTagger.cfg", ".");
+
+        //run top tager
+        //std::cout << "Run TopTagger" << std::endl;
+        tt.runTagger(constituents);
+
+        //get tagger results 
+        //std::cout << "Get TopTagger results" << std::endl;
+        const TopTaggerResults& ttr = tt.getResults();
+
+        //print top properties
+        //get reconstructed tops
+        const std::vector<TopObject*>& tops = ttr.getTops();
+        
+        // --- version using topsByType map
+        // --- requires different loops and reordering of AllTops
+        //
+        // std::map<TopObject::Type, std::vector<TopObject*>> topsByType = ttr.getTopsByType();
+        // const std::vector<TopObject*>& MergedTops     = topsByType[TopObject::Type::MERGED_TOP];
+        // const std::vector<TopObject*>& SemiMergedTops = topsByType[TopObject::Type::SEMIMERGEDWB_TOP];
+        // const std::vector<TopObject*>& ResolvedTops   = topsByType[TopObject::Type::RESOLVED_TOP];
+        // const std::vector<TopObject*>& Ws             = topsByType[TopObject::Type::MERGED_W];
+        // nMergedTops = MergedTops.size();
+        // nSemiMergedTops = SemiMergedTops.size();
+        // nResolvedTops = ResolvedTops.size();
+        // nAllTops = nMergedTops + nSemiMergedTops + nResolvedTops;
+        // nWs = Ws.size();
+        // for (const auto* top : MergedTops)      MergedTopsTLV->push_back(top->p());
+        // for (const auto* top : SemiMergedTops)  SemiMergedTopsTLV->push_back(top->p());
+        // for (const auto* top : ResolvedTops)    ResolvedTopsTLV->push_back(top->p());
+        // for (const auto* top : Ws)              WTLV->push_back(top->p());
+        //
+        // --- version using topsByType map
+
+        std::cout << "----------------------------------------------------------------------" << std::endl;
+        unsigned int topidx = 0;
+        for(const TopObject* top : tops)
+        {
+            TopObject::Type type = top->getType();
+            //if (tops.size() > 1) std::cout << "  top type: " << type << std::endl;
+            //std::cout << "  top type: " << type << std::endl;
+            
+            //std::cout << "  top type: " << type << std::endl;
+            if (   type == TopObject::Type::MERGED_TOP
+                || type == TopObject::Type::SEMIMERGEDWB_TOP
+                || type == TopObject::Type::RESOLVED_TOP)
+            {
+                AllTopsTLV->push_back(top->p());
+                if (type == TopObject::Type::MERGED_TOP)        MergedTopsTLV->push_back(top->p());
+                if (type == TopObject::Type::SEMIMERGEDWB_TOP)  SemiMergedTopsTLV->push_back(top->p());
+                if (type == TopObject::Type::RESOLVED_TOP)      ResolvedTopsTLV->push_back(top->p());
+                if (type == TopObject::Type::MERGED_W)          WTLV->push_back(top->p());
```

Comment:
I do not understand the logic here ... above the if does not include MERGED_W, so this will never be true?